server/routes/referrals/rarDaysForm.test.ts

```diff
@@ -0,0 +1,152 @@
+import { Request } from 'express'
+import RarDaysForm from './rarDaysForm'
+import serviceCategoryFactory from '../../../testutils/factories/serviceCategory'
+
+describe(RarDaysForm, () => {
+  const serviceCategory = serviceCategoryFactory.build({ name: 'accommodation' })
+
+  describe('isValid', () => {
+    describe('when there are no errors in the form', () => {
+      it('returns true', async () => {
+        const form = await RarDaysForm.createForm({ body: { 'using-rar-days': 'no' } } as Request, serviceCategory)
+
+        expect(form.isValid).toBe(true)
+      })
+    })
+
+    describe('when there are errors in the form', () => {
+      it('returns false', async () => {
+        const form = await RarDaysForm.createForm({ body: {} } as Request, serviceCategory)
+
+        expect(form.isValid).toBe(false)
+      })
+    })
+  })
+
+  describe('errors', () => {
+    describe('when no answer is selected for using-rar-days', () => {
+      it('gives an error for using-rar-days', async () => {
+        const form = await RarDaysForm.createForm({ body: {} } as Request, serviceCategory)
+
+        expect(form.errors).toEqual([
+          { field: 'using-rar-days', message: 'Select yes if you are using RAR days for the accommodation service' },
+        ])
+      })
+    })
+
+    describe('when choosing no for using-rar-days', () => {
+      it('gives no error', async () => {
+        const form = await RarDaysForm.createForm({ body: { 'using-rar-days': 'no' } } as Request, serviceCategory)
+
+        expect(form.errors).toBeNull()
+      })
+    })
+
+    describe('when choosing yes for using-rar-days and providing a valid value for maximum-rar-days', () => {
```

Comment:
This `describe` might need to be cut down (states "providing a valid value for max rar days" but then there are some different states, e.g. blank value in nested `describe`s)

Reply:
Good spot, thanks. Fixed it.